cmd/kwild/server/migration.go

```diff
@@ -0,0 +1,212 @@
+package server
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"crypto/sha256"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/kwilteam/kwil-db/cmd/kwil-admin/nodecfg"
+	"github.com/kwilteam/kwil-db/cmd/kwild/config"
+	"github.com/kwilteam/kwil-db/common/chain"
+	commonCfg "github.com/kwilteam/kwil-db/common/config"
+	"github.com/kwilteam/kwil-db/core/client"
+	"github.com/kwilteam/kwil-db/core/log"
+	"github.com/kwilteam/kwil-db/core/types"
+	"github.com/kwilteam/kwil-db/internal/abci/cometbft"
+	"github.com/kwilteam/kwil-db/internal/statesync"
+)
+
+// The migrationClient type is responsible for:
+//   - Polling the old chain to retrieve the genesis state required for migration.
+//   - Downloading the genesis snapshot from the old chain and saves it under the
+//     root directory of the node.
+//   - Updating the genesis configuration such as app hash, validators and migration settings.
+//   - Updating the kwild configuration with the snapshot file path and migrations listener extension.
+
+const (
+	defaultPollFrequency = time.Second * 30
+)
+
+type migrationClient struct {
+	// listenAddress is the old chain's listen address to retrieve the genesis state
+	listenAddress string
+
+	snapshotFileName string
+
+	clt        *client.Client
+	kwildCfg   *commonCfg.KwildConfig
+	genesisCfg *chain.GenesisConfig
+
+	logger log.Logger
+}
+
+func PrepareForMigration(ctx context.Context, kwildCfg *commonCfg.KwildConfig, genesisCfg *chain.GenesisConfig, logger log.Logger) (*commonCfg.KwildConfig, *chain.GenesisConfig, error) {
```

Comment:
Please adds docs to all structs and functions in this file. Just something small so we know what is going on